elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchFlowStage.scala

```diff
@@ -79,28 +104,52 @@ class ElasticsearchFlowStage[T, R](
       private def handleSuccess(): Unit =
         completeStage()
 
-      private def handleResponse(args: (Seq[IncomingMessage[T]], Response)): Unit = {
+      private def handleResponse(args: (Seq[IncomingMessage[T, C]], Response)): Unit = {
         val (messages, response) = args
         val responseJson = EntityUtils.toString(response.getEntity).parseJson
 
         // If some commands in bulk request failed, pass failed messages to follows.
         val items = responseJson.asJsObject.fields("items").asInstanceOf[JsArray]
-        val failed = items.elements.zip(messages).flatMap {
+        val messageResults = items.elements.zip(messages).map {
           case (item, message) =>
-            item.asJsObject.fields("index").asJsObject.fields.get("error").map { _ =>
-              message
+            item.asJsObject.fields("index").asJsObject.fields.get("error") match {
+              case Some(errorMessage) => (message, false)
+              case None => (message, true)
             }
         }
 
-        if (failed.nonEmpty && settings.retryPartialFailure && retryCount < settings.maxRetry) {
+        val successMsgs = messageResults.filter(_._2).map(_._1)
+        val failedMsgs = messageResults.filter(!_._2).map(_._1)
```

Comment:
Classifying successed messages and failed messages at once using `groupBy` might be better?

Reply:
I'm not sure how to improve this one using groupBy